### PR TITLE
Added: LightningStrikeEvent

### PR DESCRIPTION
**Pull request by anna-is-cute**
The most important event by far. If this should be done for the effect as well, let me know, and I'll update it.

### CODE REVIEW

**Comment by SpaceManiac** [comment]
I know it's a little confusing, but for very simple events (i.e. one-liners) like this one prefer creating event on-site & only using `EventFactory.callEvent`. Also I think it should be fired for effects too - plugins can differentiate with `event.getLightning().isEffect()`.

**Comment by anna-is-cute** [comment]
Damn, that's how I had it, but I saw other events and changed it. Will fire for effects, too, then.

**Comment by turt2live** [comment]
@jkcclemens this PR does not currently merge. Please fix that ASAP!

**Comment by anna-is-cute** [comment]
Am currently grumbling about other PRs making my PR not able to merge. Will fix later tonight, hopefully.

**Comment by anna-is-cute** [comment]
This can now be merged.

File: src/main/java/net/glowstone/GlowWorld.java [hunk 1]
```diff
@@ -1000,14 +1001,23 @@ public LivingEntity spawnCreature(Location loc, CreatureType type) {
         return (LivingEntity) spawn(loc, type.getEntityClass());
     }
 
+    private GlowLightningStrike strikeLightningFireEvent(final Location loc, final boolean effect) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/GlowWorld.java:1004]: I don't think any of these variables need to be `final`, but that might just be me being too nitpicky.

**Comment by turt2live** [comment]
Looks good otherwise. Passing this off to SpaceManiac for review/pull.

File: src/main/java/net/glowstone/GlowWorld.java (hunk 1 above)
Comment (anna-is-cute) [inline src/main/java/net/glowstone/GlowWorld.java:1004]: Just my code style. Not harming anything, is it?

Comment (JeromSar) [inline src/main/java/net/glowstone/GlowWorld.java:1004]: @jkcclemens This should probably be in `EventFactory` named `onLightningStrike`

Comment (anna-is-cute) [inline src/main/java/net/glowstone/GlowWorld.java:1004]: No, it shouldn't. Look at the PR conversation. I had it like that originally.

**Comment by JeromSar** [comment]
Related parent: #28

File: src/main/java/net/glowstone/GlowWorld.java (hunk 1 above)
Comment (JeromSar) [inline src/main/java/net/glowstone/GlowWorld.java:1004]: @jkcclemens I see. Quite odd to events being called in many different places. That's probably not of the best interest to code clarity. Additionally, a lot of one-liners in called in `EventFactory` too. @SpaceManiac ?

**Comment by SpaceManiac** [comment]
Merged in f78a5ed416672cb4c1a683e2f22622a9978bc9e3.

**Pull request closed without merging**